### ISSUE TITLE
Improve documentation for supported samplers

### ISSUE THREAD

**Issue by rkondrashov** [open-telemetry/opentelemetry-java-instrumentation#2245]
Please improve documentation for supported samplers, now it's not always clear what each sampler does exactly.
See documentation for Python SDK - https://opentelemetry-python.readthedocs.io/en/stable/sdk/trace.sampling.html, it's already done there.

**Comment by iNikem** [comment]
@rkondrashov I think this documentation should be in https://github.com/open-telemetry/opentelemetry-java, what do you think?

**Comment by rkondrashov** [comment]
I agree, it would be better to document samplers in https://github.com/open-telemetry/opentelemetry-java and add a link to the "Trace config" section of the `agent-config.md`.

**Comment by iNikem** [comment]
Do you mind then to move this issue to that repository?

**Comment by trask** [comment]
@anuraaga and @jkwatson have transfer powers between the two repos
